pkgs/development/compilers/heptagon/default.nix

```
@@ -0,0 +1,49 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, makeWrapper
+, ocamlPackages
+}:
+
+stdenv.mkDerivation rec {
+  pname = "heptagon";
+  version = "1.05.00";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.inria.fr";
+    owner = "synchrone";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-b4O48MQT3Neh8a1Z5wRgS701w6XrwpsbSMprlqTT+CE=";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  buildInputs = with ocamlPackages; [
+    ocaml
+    findlib
+    menhir
+    menhirLib
+    camlp4
+    ocamlbuild
+  ];
+
+  propagatedBuildInputs = with ocamlPackages; [
+    lablgtk
+    ocamlgraph
+  ];
+
+  postFixup = with ocamlPackages; ''
+    wrapProgram $out/bin/hepts \
+      --prefix CAML_LD_LIBRARY_PATH : "${lablgtk}/lib/ocaml/${ocaml.version}/site-lib/lablgtk2"
+  '';
+
+  meta = with lib; {
```

Since the name of the package differs from provided executables, doing something like this
```suggestion
  meta = with lib; {
    mainProgram = "heptc";
```
would allow us to run the package with `nix run nixpkgs#heptagon`. Assuming that the main executable should be `heptc` and not `hepts`. This is just my personal preference, if you think that's not needed, feel free to ignore this comment.

---

Done. Thanks!